pages/partials/studentGroupControls.ejs

```diff
@@ -28,10 +28,33 @@
     Failed to create the group. It is already taken. Please try another one.
   </div>
 <% } %>
+
 <%- include('../shared/groupWorkInitial.ejs', {permissions: permissions}); %>
 <% } %>
 <% } else {%>
   <%- include('../shared/groupWorkInfoContainer.ejs', {permissions: permissions, group_info: group_info}); %>
+  <% if (usingGroupRoles && !rolesAreBalanced) { %>
+    <div class="alert alert-danger my-2" role="alert">
+      A user has too many roles. Every student should be assigned to exactly <strong>one</strong> role with group size <strong><%= groupsize %></strong> 
+    </div>
+  <% } %>
+  <% if (usingGroupRoles && can_view_role_table) { %>
+    <% if (typeof validationErrors !== 'undefined' && validationErrors.length > 0) { %>
+      <% validationErrors.forEach(function ({ role_name, minimum, maximum, num_assigned }) { %>
+        <div class="alert alert-danger my-2" role="alert">
+          <%= role_name %>
+        <% if (num_assigned < minimum) { %>
+          has too few assignments (found  
+          <% } else { %>
+          has too many assignments (found
+          <% } %>
+          <%= num_assigned %> but expected <%= minimum %> minimum and <%= maximum %> maximum.)
```

Comment:
```suggestion
          <%= num_assigned %> but expected between <%= minimum %> and <%= maximum %>.)
```

Consider special-casing the case where `minimum === maximum` - regardless of what phrasing is used, it'd be weird to read "expected 1 minimum and 1 maximum" or "expected between 1 and 1".